File: templates/fsharp/app-mvvm/ViewModels/ViewModelBase.fs
namespace AvaloniaAppTemplate.ViewModels

#if (CommunityToolkitChosen)
open CommunityToolkit.Mvvm.ComponentModel
#elif (ReactiveUIToolkitChosen)
open ReactiveUI
#endif

type ViewModelBase() =
#if (CommunityToolkitChosen)
    inherit ObservableObject()
#elif (ReactiveUIToolkitChosen)
    inherit ReactiveObject()
#endif

File: templates/csharp/app-mvvm/ViewModels/ViewModelBase.cs
#if (CommunityToolkitChosen)
using CommunityToolkit.Mvvm.ComponentModel;
#elif (ReactiveUIToolkitChosen)
using ReactiveUI;
#endif

namespace AvaloniaAppTemplate.ViewModels;

#if (CommunityToolkitChosen)
public abstract class ViewModelBase : ObservableObject
#elif (ReactiveUIToolkitChosen)
public abstract class ViewModelBase : ReactiveObject
#endif
{
}
